src/index.ts

```diff
@@ -84,7 +84,15 @@ type ReadonlyEquivalent<X, Y> = Extends<
 export type Extends<L, R> = IsNever<L> extends true ? IsNever<R> : [L] extends [R] ? true : false
 export type StrictExtends<L, R> = Extends<DeepBrand<L>, DeepBrand<R>>
 
-export type Equal<Left, Right> = And<[StrictExtends<Left, Right>, StrictExtends<Right, Left>]>
+type StrictEqual<L, R> =
+  (<T>() => T extends (L & T) | T ? true : false) extends
```

Comment:
Is there a link to a stack overflow/blog post/docs page that explains this trick or something similar? And/or an explanation of why we can't just use `T`. It's befuddling(-looking) enough to warrant a comment.